lib/cache/store.go

```diff
@@ -77,7 +78,7 @@ func (s *store[T, I]) len() int {
 func (s *store[T, I]) get(index I, key string) (T, error) {
 	t, ok := s.cache.Get(index, key)
 	if !ok {
-		return t, trace.NotFound("no value for key %q in index %v", key, index)
+		return t, trace.NotFound("%v %q does not exist", reflect.TypeFor[T](), key)
```

Comment:
Is this error user facing? Can we get to a resource kind string somehow?

Reply:
That's going to be annoying for 153-style resources - is that better or worse than just requiring a user-friendly name for error messages in `store`?

...also, we're hitting this error message because failed to find the `T`, and almost all of our resources don't have a "static" `GetKind` method, they read the kind from the value. 😬